src/Microsoft.Framework.ConfigurationModel.Json/JsonConfigurationExtension.cs

```diff
@@ -1,12 +1,37 @@
 // Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
+using System.IO;
+
 namespace Microsoft.Framework.ConfigurationModel
 {
     public static class JsonConfigurationExtension
     {
         public static IConfigurationSourceRoot AddJsonFile(this IConfigurationSourceRoot configuration, string path)
         {
+            if (!Path.IsPathRooted(path))
+            {
+                if (configuration.BasePath == null)
+
+                {
+                    throw new InvalidOperationException(
+                        string.Format("The {0} is not rooted and the {1} is null in this {2}",
+                            nameof(path),
+                            nameof(configuration.BasePath),
+                            typeof(IConfigurationSourceRoot).Name));
+                }
+                else
+                {
+                    path = Path.Combine(configuration.BasePath, path);
+                }
+            }
+
+            if (!File.Exists(path))
```

Comment:
nit: could combine this with existing parameter checks in the `JsonConfigurationSource` constructor